internal/utils/persistence/src/main/java/org/eclipse/ditto/internal/utils/persistence/mongo/MongoClientWrapper.java

```diff
@@ -492,12 +501,21 @@ private static SSLContext createAndInitSslContext() throws NoSuchAlgorithmExcept
         }
 
         private void applyAwsIamRoleSettings() {
-            final AwsCredentialsProvider credentialsProvider = getAwsCredentialsProvider();
-            final AwsCredentials credentials = credentialsProvider.resolveCredentials();
-            final MongoCredential credential = MongoCredential.createAwsCredential(
-                    credentials.accessKeyId(),
-                    credentials.secretAccessKey().toCharArray()
-            );
+            final StsClient stsClient = StsClient.builder().build();
+
+            final AwsSessionCredentials tempCredentials = (AwsSessionCredentials) StsAssumeRoleCredentialsProvider.builder()
+                    .stsClient(stsClient)
+                    .refreshRequest(req -> req.roleArn(awsRoleArn).roleSessionName("sessionName"))
```

Comment:
@hu-ahmed isn't the role session name something which also should be configurable?
The string constant feels wrong here..

Reply:
The RoleSessionName is required by AWS and serves as an identifier for the assumed role session. While it can be any string meeting the AWS constraints, it's often used for logging and auditing purposes. 
[aws doc](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html)
I can make this configurable in the code or set it as a fixed value like "dittoSession." Let me know your preference!

Reply:
Configurable with default "Ditto" would be great 😉